akka-remote/src/test/scala/akka/remote/artery/LargeMessagesStreamSpec.scala

```diff
@@ -0,0 +1,155 @@
+    }
+  }
+}
+
+class LargeMessagesStreamSpec extends WordSpec with ShouldMatchers with ScalaFutures {
+  import LargeMessagesStreamSpec._
+
+  val config = ConfigFactory.parseString(
+    s"""
+      akka {
+        loglevel = ERROR
+        actor {
+          provider = "akka.remote.RemoteActorRefProvider"
+        }
+        remote.artery {
+          enabled = on
+          hostname = localhost
+          port = 0
+          large-message-destinations = [
+            "/user/large"
+          ]
+        }
+      }
+
+    """)
+
+  "The large message support" should {
+
+    "not affect regular communication" in {
+      val systemA = ActorSystem("systemA", config)
+      val systemB = ActorSystem("systemB", config)
+
+      try {
+        val senderProbeA = TestProbe()(systemA)
+        val senderProbeB = TestProbe()(systemB)
+
+        // start actor and make sure it is up and running
+        val large = systemB.actorOf(Props(new EchoSize), "regular")
+        large.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        // communicate with it from the other system
+        val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val rootB = RootActorPath(addressB)
+        val largeRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "regular"))
+        largeRemote.tell(Ping(), senderProbeA.ref)
+        senderProbeA.expectMsg(Pong(0))
+
+        // flag should be cached now
+        largeRemote.asInstanceOf[RemoteActorRef].cachedLargeMessageDestinationFlag should ===(RegularDestination)
+
+      } finally {
+        TestKit.shutdownActorSystem(systemA)
+        TestKit.shutdownActorSystem(systemB)
+      }
+    }
+
+    "pass small regular messages over the large-message stream" in {
+      val systemA = ActorSystem("systemA", config)
+      val systemB = ActorSystem("systemB", config)
+
+      try {
+        val senderProbeA = TestProbe()(systemA)
+        val senderProbeB = TestProbe()(systemB)
+
+        // start actor and make sure it is up and running
+        val large = systemB.actorOf(Props(new EchoSize), "large")
+        large.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        // communicate with it from the other system
+        val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val rootB = RootActorPath(addressB)
+        val largeRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "large"))
+        largeRemote.tell(Ping(), senderProbeA.ref)
+        senderProbeA.expectMsg(Pong(0))
+
+        // flag should be cached now
+        largeRemote.asInstanceOf[RemoteActorRef].cachedLargeMessageDestinationFlag should ===(LargeDestination)
+
+      } finally {
+        TestKit.shutdownActorSystem(systemA)
+        TestKit.shutdownActorSystem(systemB)
+      }
+    }
+
+    "allow for normal communication while simultaneously sending large messages" in {
```

Comment:
we should probably create a latency spec that tests the regular message channel latency distribution while large messages are transmitted in the background.